### PR TITLE
docs: Document how to backfill a cron workflow

### PR DESCRIPTION
**Pull request by alexec**
Checklist:

* [x] Either (a) I've created an [enhancement proposal](https://github.com/argoproj/argo/issues/new/choose) and discussed it with the community, (b) this is a bug fix, or (c) this is a chore.
* [x] The title of the PR is (a) [conventional](https://www.conventionalcommits.org/en/v1.0.0/), (b) states what changed, and (c) suffixes the related issues number. E.g. `"fix(controller): Updates such and such. Fixes #1234"`.  
* [x] I've signed the CLA.
* [x] I have written unit and/or e2e tests for my change. PRs without these are unlikely to be merged.
* [x] My builds are green. Try syncing with master if they are not. 
* [x] My organization is added to USERS.md.

### CODE REVIEW

**Comment by alexec** [comment]
This can be merged as soon as approved.

File: examples/cron-backfill.yaml [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+            name: job-{{inputs.parameters.date}}
+          spec:
+            entrypoint: main
+            templates:
+              - name: main
+                steps:
+                  - - name: start-job
+                      templateRef:
+                        name: job
+                        template: main
```
Comment (simster7) [inline examples/cron-backfill.yaml:87]: Should we be passing down the `date` argument here? As it is done in the `v2` example?

```yaml
        - - name: create-job
            templateRef:
              name: job
              template: main
            arguments:
              parameters:
                - name: date
                  value: "{{item}}"
            withSequence:
              # run for the 19th to the 21st
              start: "19"
              end: "21"
              format: "2020-05-%02X"
```

File: examples/cron-backfill.yaml [hunk 2]
```diff
@@ -0,0 +1,114 @@
+# This is an example of back-filling day for a daily cron workflow.
```
Comment (simster7) [inline examples/cron-backfill.yaml:1]: A link to `docs/cron-backfill.md` would be useful here for users who find this example independently from the docs.

File: examples/cron-backfill.yaml [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+            default: yesterday
+      script:
+        image: docker/whalesay:latest
+        command:
+          - sh
+        source: |
+          date="{{inputs.parameters.date}}"
+          if [ $date = yesterday ]; then
+            date=$(date -d yesterday +%Y-%m-%d)
+          fi
```
Comment (simster7) [inline examples/cron-backfill.yaml:26]: Shouldn't the default action here (i.e. if no inputs are passed in) to do nothing?

File: examples/cron-backfill.yaml (hunk 1 above)
Comment (alexec) [reply examples/cron-backfill.yaml:87]: yes - we should

File: examples/cron-backfill.yaml (hunk 3 above)
Comment (alexec) [reply examples/cron-backfill.yaml:26]: no, I think by default it needs to run as if it was a normal daily cron

File: examples/cron-backfill.yaml (hunk 2 above)
Comment (alexec) [reply examples/cron-backfill.yaml:1]: done

File: examples/cron-backfill.yaml (hunk 1 above)
Comment (alexec) [reply examples/cron-backfill.yaml:92]: done

File: examples/cron-backfill.yaml (hunk 3 above)
Comment (007vasy) [reply examples/cron-backfill.yaml:27]: @alexec the argo list shows an error on this, can you point me to the updated one if there is any?

**Pull request merged**